packages/api/cms-api/src/dam/files/dto/file.input.ts

```diff
@@ -58,13 +58,13 @@ export class LicenseInput {
     author?: string;
 
     @Field(() => Date, { nullable: true })
-    @Transform(({ value }) => new Date(value))
+    @Transform(({ value }) => (value === null || value === undefined ? null : new Date(value)))
```

Comment:
The type is `undefined`, so it should probably transform to undefined, right?

Reply:
Well the type is either "RIGHTS_MANAGED" or "ROYALTY_FREE"

Noticed this bug in my project. Could fix it there by doing a `?? undefined`, but should be fixed in the package to be straightforward in the project.

![Screenshot 2024-12-02 at 10 20 29](https://github.com/user-attachments/assets/2e3903b6-00bc-402f-a31e-de87ce9b3ea6)

Reply:
I was talking about the type of the duration fields, which are types as Date or undefined

Reply:
In the database it is saved as null or Date ... so should it actually be nullable or Date in the dto? 🤔

Reply:
> so should it actually be nullable or Date in the dto?

Either this, or both undefined and null

Reply:
might be non breaking to change IsOptional to IsNullable. did not see it used for any of our projects using the GQLInput. should only need the additional change in the EditFile.tsx

![Screenshot 2024-12-02 at 14 04 59](https://github.com/user-attachments/assets/6d9a076a-cc4f-42c3-89be-ba041264ab1c)

![Screenshot 2024-12-02 at 14 05 17](https://github.com/user-attachments/assets/b9ea6086-bc8c-4d0f-ae32-31377953c9a6)

Reply:
I think if we fix this we should should also change it in the entity to `durationFrom: Date | null`.

Searched for `| null;` but it is apparently not used in a single entity.

![Screenshot 2024-12-03 at 15 53 14](https://github.com/user-attachments/assets/60a5fdfd-298d-4b91-80fa-e0803917e045)

Reply:
c31f7e1580161a3264ee5df89d146eae31d54778